cmd/zfs_object_agent/zettacache/src/block_allocator.rs

```diff
@@ -579,7 +660,12 @@ pub struct BlockAllocator {
     coverage: Extent,
     slab_size: u32,
 
+    // XXX: Explain
+    // + also generation bump
     spacemap: SpaceMap,
+    spacemap_next: SpaceMap,
+    condensed_slabs: u64,
```

Comment:
I guess we will be condensing all the time; maybe that's fine?  If not then the next spacemap should be None, as should the next slab to condense.  One way to do that would be:
```suggestion
    condensing: Option<(SpaceMap, SlabId)>,
```

But assuming we do want to be condensing all the time, I would recommend `next_slab_to_condense: SlabId`